app/src/main/java/com/github/polypoly/app/ui/menu/lobby/UniqueGameLobbyCodeGenerator.kt

```diff
@@ -13,12 +13,15 @@ import kotlin.random.Random
  */
 class UniqueGameLobbyCodeGenerator(private val codeLength: Int = GAME_LOBBY_CODE_LENGTH) {
 
-    fun generateUniqueGameLobbyCode(): String {
-        var code: String
-        do {
-            code = generateCode()
-        } while (!codeIsValid(code))
-        return code
+    fun generateUniqueGameLobbyCode(): CompletableFuture<String> {
+        val code = generateCode()
+        return codeIsUsed(code).thenCompose { isUsed ->
+            if(isUsed) {
+                generateUniqueGameLobbyCode()
+            } else {
+                CompletableFuture.completedFuture(code)
+            }
+        }
```

Comment:
Nitpick, but usually the way this is handled is using something like 'previousUID++', to avoid relying on randomness and ensuring quick and guaranteed result (obviously limited by MAX_INT)

Reply:
As this is @Kaz-ookid 's work, I'll leave it like this but I strongly agree with you

Reply:
I agree, thanks for the advice I did not think about it. However this works fine for now, as the code is way bigger than needed seeing how many of those codes we will have to produce. This will be to be though about again if we get thousands of players daily i guess haha